src/backtrack/Main.java

```diff
@@ -0,0 +1,45 @@
+package backtrack;
+
+import java.util.*;
+// This is a branch of master
+public class Main {
+
+//this is the main class where the algorithm will be executed from.
+    public static void main(String[] args) {
+        ReadMaze rm = new ReadMaze();
+        Scanner sc = new Scanner(System.in);
+        String[][] maze;
+        Set<String> keys = new HashSet<>();
+        keys.add("#");
+        keys.add("S");
+        keys.add("F");
+
+        System.out.println("Input the filename with the extention");
+        rm.setFilePath(sc.nextLine());
+        maze = rm.readMaze();
+        TraverseMaze tm = new TraverseMaze(maze);
+        for (int i = 0; i < maze.length; i++) {
+            for (int j = 0; j < maze[i].length; j++) {
+                System.out.print(maze[i][j] + " ");
+            }
+            System.out.println();
+        }
+        //find the path
+        tm.move(rm.getStartPos());
+        for (int[][] node : tm.getPathStack()) {
```

Comment:
After you find the path or not, you are adding the path to your maze and replacing 0 with empty spaces 
This is should be part of the main method but part of the TraverseMaze class, because it is not the responsibility of this main class to finish the job of it. Ideally, it should call the maze reader and pass the result to the traverse to find the solution and print it.

See SOLID principles: https://stackify.com/solid-design-principles/